### PR TITLE
Run RISC-V Compliance without the pytest-based test scripts

### PR DESCRIPTION
**Pull request by imphil**
As part of ongoing work to refactor the system tests to use pytest not only as helper library, but as test runner, some helper scripts will be removed. These scripts are still used in the way we run the RISC-V Compliance test suite. Rework the makefiles to call the necessary tools directly.

Notes:
* The commit structure will be slightly reworked after review of this PR is complete to fold the vendored patch into an existing patch.
* The way RISC-V compliance is run isn't exactly beautiful, but it does the job.
* See also the commit message in the patch for more details what changed in RISC-V compliance.

### CODE REVIEW

**Comment by imphil** [comment]
A note on CI runtimes: The setup in this PR matches the parallelism we had before. I did experiment with increasing the number of threads within the simulation, but decreasing the number of parallel test runs. Unfortunately, the machines we get from Azure differ widely between runs (by a factor of 2 at times), making a comparison of runtimes impossible. I therefore chose to leave things as they were before, we can look at that topic again another time.

**Review by GregAC — APPROVED** [review]
Thanks for sorting this out, couple of comments but happy with it as is if you've got other things to be getting on with.

> A note on CI runtimes: The setup in this PR matches the parallelism we had before. I did experiment with increasing the number of threads within the simulation, but decreasing the number of parallel test runs. Unfortunately, the machines we get from Azure differ widely between runs (by a factor of 2 at times), making a comparison of runtimes impossible. I therefore chose to leave things as they were before, we can look at that topic again another time.

Indeed I looked at that before when setting it up. Parallel runs within the same job didn't seem to gain you that much, explicit splitting on tests between jobs is what gave the major boost. As the rv32i suite is far bigger than the rest to get better performance (reliably) you'd need to alter the makefiles to allow sub-suite runs so you can execute say half of rv32i or manually create new suites that are parts of rv32i.

I note the CI run has a bunch of these messages:

> dmi0: Failed to bind socket: Address already in use (98)
dmi0: Unable to create TCP server on port 44853

Any idea what those are about?

File: sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/Makefile.include [hunk 1]
```diff
@@ -3,51 +3,46 @@
...
-        $(PYTEST_OPT) \
-        --log $(<).uart.log; \
-        grep -o 'SIG: [a-zA-Z0-9_]*' $(<).uart.log | sed 's/SIG: //' \
-        > $(*).signature.temp.output; \
-        tr '[:upper:]' '[:lower:]' < $(*).signature.temp.output > $(*).signature.output;
-
+ifeq ($(OT_TARGET),sim_verilator)
+    MAKEARG = SIM=1
+    CARG = -DSIMULATION=1
+    RUN_TARGET = \
```
[the remark this answers is not in the document]
Comment (imphil) [reply sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/Makefile.include:26]: Sure! I added code to dump the UART output when running with `make VERBOSE=1`, and updated the CI configuration to use this setting.

**Comment by imphil** [comment]
> I note the CI run has a bunch of these messages:
>
>    dmi0: Failed to bind socket: Address already in use (98)
>    dmi0: Unable to create TCP server on port 44853
>
> Any idea what those are about?

These warnings have been present before, but were not as visible.

Its an artifact of running more than one Verilator simulation in parallel, which is actually not really supported. Most of the DPI peripherals assume there's only a single simulation running, and make use of global system-wide resources that way (e.g. by choosing a fixed port).

That's why I tried to change things to run only a single simulation in parallel (number of make jobs == 1), but speed up the simulation itself by configuring Verilator to create a threaded simulation. That, as it turns out, probably gives us slightly slower end-to-end runtime of RISC-V compliance in Azure CI (probably due to the time it takes to compile the test software snippets, and to compare its output). I wasn't really able to get conclusive numbers on that, however, so I left things as they were for now.

**Pull request merged**